### ISSUE TITLE
spack compiler find adds python unicode strings

### ISSUE THREAD

**Issue by adamjstewart** [spack/spack#3842]
Just noticed this interesting behavior. I deleted my `compilers.yaml` and `spack compiler find` created a new one that looks like:
```yaml
compilers:
- compiler:
    environment: {}
    extra_rpaths: []
    flags: {}
    modules: []
    operating_system: sierra
    paths:
      cc: !!python/unicode '/Library/Developer/CommandLineTools/usr/bin/clang'
      cxx: !!python/unicode '/Library/Developer/CommandLineTools/usr/bin/clang++'
      f77: /usr/local/bin/gfortran
      fc: /usr/local/bin/gfortran
    spec: clang@8.1.0-apple
    target: x86_64
```
I'm going to investigate why it adds these `!!python/unicode` strings. Just wanted to report it so I can keep track of the bug.

**Comment by adamjstewart** [comment]
Having a little trouble tracking this one down. Note that this bug doesn't seem to affect Spack when run with Python 3, only with Python 2.

**Comment by scheibelp** [comment]
In my case (on linux, w/ python 2.7.10) ```spack compiler find``` does not add these tags

My guess would be that on sierra the file paths are read as unicode strings and then persist as such until written out by spack_yaml (which decides to preserve that in the representation)

I think this issue could be resolved by controlling how spack_yaml writes out strings. Is this creating problems? I could start working on this Monday

**Comment by adamjstewart** [comment]
It isn't causing any problems, just looks sketchy. No rush.

**Comment by adamjstewart** [comment]
@tgamblin @scheibelp This bug was fixed by #5077, but it looks like #5077 was reverted by #10186. Any idea how to handle this properly?

**Comment by alalazo** [comment]
We don't support running with Python 2 anymore.
